### ISSUE TITLE
Type "Literal[0]" cannot be assigned to type "int"

### ISSUE THREAD

**Issue by Tpinion** [microsoft/pyright#4415]
**Describe the bug**
There is no error on following code when using pyright 1.1.286.
```python
a: int = 1
while True:  # Any loop.
    while a >= 0:
        a -= 1
    a = max(0, a)
```
It raises following error when using pyright 1.1.287 on same code.
```
 error: Argument of type "int" cannot be assigned to parameter "__arg2" of type "SupportsRichComparisonT@max" in function "max"
    Type "Literal[0]" cannot be assigned to type "int" (reportGeneralTypeIssues)
```
**To Reproduce**
1. create `test.py`
2. paste next code.
```python
a: int = 1
while True:  # Any loop.
    while a >= 0:
        a -= 1
    a = max(0, a)
```
3. execute `pyright`

**Expected behavior**
If I does a forced type conversion on zero,I can avoid this error hint. But it seems very strange.
```python
a: int = 1
while True:  # Any loop.
    while a >= 0:
        a -= 1
    a = max(int(0), a)
```

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/27395365/210949102-00549492-51b5-457e-8c73-0eb33f0af388.png)

**Additional context**
This is the minimum sample code to repeat the problem, because raw code has much irrelevant logic.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.288, which I just published. It will also be included in a future release of pylance.
